core/indigo-core/molecule/src/molecule_auto_loader.cpp

```diff
@@ -221,6 +240,22 @@ void MoleculeAutoLoader::_loadMolecule(BaseMolecule& mol, bool query)
         }
     }
 
+    // check for CDX format
+    {
+        if (local_scanner->findWord("VjCD0100"))
+        {
+            MoleculeCdxLoader loader(*local_scanner);
+            loader.stereochemistry_options = stereochemistry_options;
+            if (query)
+                throw Error("CDX queries not supported yet");
+            loader.loadMolecule(mol.asMolecule());
+
+            properties.copy(loader.properties);
+
+            return;
+        }
+    }
```

Comment:
Why do we move it upper?

Reply:
Binary formats should be handled first to avoid text format parsers of trying to parse binary.